sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -296,6 +296,17 @@ object SQLConf {
       .longConf
       .createWithDefault(250 * 1024 * 1024)
 
+  val HIVE_SCHEMA_INFERENCE_MODE = buildConf("spark.sql.hive.schemaInferenceMode")
+    .doc("Configures the action to take when a case-sensitive schema cannot be read from a Hive " +
+      "table's properties. Valid options include INFER_AND_SAVE (infer the case-sensitive " +
+      "schema from the underlying data files and write it back to the table properties), " +
+      "INFER_ONLY (infer the schema but don't attempt to write it to the table properties) and " +
+      "NEVER_INFER (fallback to using the case-insensitive metastore schema instead of inferring).")
+    .stringConf
+    .transform(_.toUpperCase())
+    .checkValues(Set("INFER_AND_SAVE", "INFER_ONLY", "NEVER_INFER"))
+    .createWithDefault("INFER_AND_SAVE")
```

Comment:
INFER_ONLY seems better to me as a default since it avoid throwing exceptions. To avoid silent performance degradation in this case, perhaps we can log a warning pointing to this config? Alternatively, I could see INFER_AND_SAVE as the default if we caught the write permission errors.